src/lib.rs

```diff
@@ -0,0 +1,47 @@
+// Copyright 2018 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License.  You may obtain a copy
+// of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+// License for the specific language governing permissions and limitations
+// under the License.
+
+extern crate fuse;
+#[macro_use] extern crate log;
+
+use std::ffi::OsStr;
+use std::io;
+use std::path::Path;
+
+/// FUSE file system implementation of sandboxfs.
+struct SandboxFS {
+}
+
+impl SandboxFS {
+    /// Creates a new `SandboxFS` instance.
+    fn new() -> SandboxFS {
+      SandboxFS {}
+    }
+}
+
+impl fuse::Filesystem for SandboxFS {
+}
+
+/// Mounts a new sandboxfs instance on the given `mount_point`.
+pub fn mount(mount_point: &Path) -> io::Result<()> {
+    let options = ["-o", "ro", "-o", "fsname=sandboxfs"]
+        .iter()
+        .map(|o| o.as_ref())
+        .collect::<Vec<&OsStr>>();
+    let fs = SandboxFS::new();
+    info!("Mounting file system onto {:?}", mount_point);
+    fuse::mount(fs, &mount_point, &options)
```

Comment:
I see why you say that... but I don't understand why the compiler doesn't agree with you:

```
error[E0277]: the trait bound `[u8]: std::marker::Sized` is not satisfied in `std::path::Path`
  --> src/lib.rs:44:5
   |
44 |     fuse::mount(fs, mount_point, &options)
   |     ^^^^^^^^^^^ `[u8]` does not have a constant size known at compile-time
   |
   = help: within `std::path::Path`, the trait `std::marker::Sized` is not implemented for `[u8]`
   = note: required because it appears within the type `std::path::Path`
   = note: required by `fuse::mount`
```

Reply:
I've sent https://github.com/zargony/rust-fuse/pull/108 to adjust this per your suggestion. Not sure if it's possible to do this without breaking compatibility though.

Reply:
Alternatively you could just add ?Sized to their constraint: `P: AsRef<Path>+?Sized`